### ISSUE TITLE
False positive RedundantCondition for console option with default value

### ISSUE THREAD

**Issue by gharlan** [psalm/psalm-plugin-symfony#86]
```php
    protected function configure()
    {
        $this
            ->addOption('foo', null, InputOption::VALUE_OPTIONAL, 'desc', false)
        ;
    }

    protected function execute(InputInterface $input, OutputInterface $output)
    {
        $foo = $input->getOption('foo');

        if (false !== $foo) {
            // option is set (with or without value)
        }

        // ...
    }
```

This results in a RedundantCondition "string can never contain false".
But actually the value can be false, it is the default value.
I'm using this to differentiate between "option is not used" (`false`) and "option is used without value" (`null`).

**Comment by seferov** [comment]
@gharlan thank you for the report. https://github.com/psalm/psalm-plugin-symfony/pull/92 should fix the issue
